rasa/nlu/extractors/duckling_entity_extractor.py

```diff
@@ -172,41 +177,30 @@ def _reference_time_from_message(message: Message) -> int:
                     "duckling. Error: {}".format(message.time, e)
                 )
         # fallbacks to current time, multiplied by 1000 because duckling
-        # requires the reftime in miliseconds
+        # requires the reftime in milliseconds
         return int(time.time()) * 1000
 
-    def process(self, message: Message, **kwargs: Any) -> None:
-
-        if self._url() is not None:
-            reference_time = self._reference_time_from_message(message)
-            matches = self._duckling_parse(message.get(TEXT), reference_time)
-            all_extracted = convert_duckling_format_to_rasa(matches)
-            dimensions = self.component_config["dimensions"]
-            extracted = DucklingEntityExtractor.filter_irrelevant_entities(
-                all_extracted, dimensions
-            )
-        else:
-            extracted = []
+    def process(self, messages: List[Message]) -> List[Message]:
+        """Augments the message with potentially extracted entities."""
+        if self._url() is None:
             rasa.shared.utils.io.raise_warning(
                 "Duckling HTTP component in pipeline, but no "
                 "`url` configuration in the config "
                 "file nor is `RASA_DUCKLING_HTTP_URL` "
                 "set as an environment variable. No entities will be extracted!",
                 docs=DOCS_URL_COMPONENTS + "#DucklingEntityExtractor",
             )
+            return messages
 
-        extracted = self.add_extractor_name(extracted)
-        message.set(ENTITIES, message.get(ENTITIES, []) + extracted, add_to_output=True)
+        for message in messages:
+            reference_time = self._reference_time_from_message(message)
+            matches = self._duckling_parse(message.get(TEXT), reference_time)
+            all_extracted = convert_duckling_format_to_rasa(matches)
+            dimensions = self.component_config["dimensions"]
+            extracted = self.filter_irrelevant_entities(all_extracted, dimensions)
+            extracted = self.add_extractor_name(extracted)
+            message.set(
+                ENTITIES, message.get(ENTITIES, []) + extracted, add_to_output=True
+            )
 
-    @classmethod
-    def load(
```

Comment:
If I understand correctly, since this component doesn't need to persist, `load` method also doesn't need to be overriden?

Reply:
exactly. `load` by default just points to `create`